### PR TITLE
Only use one TX for all subscriptions

### PR DESCRIPTION
**Pull request by arv**
Follow up to 4021d6d92e759478b6ac3e401879587635425d2b

### CODE REVIEW

File: lib/replicache.dart [hunk 1]
```diff
@@ -356,9 +363,17 @@ class Replicache implements ReadTransaction {
...
     yield await query(callback);
 
-    await for (final _ in _changeStreamController.stream) {
-      yield await query(callback);
-    }
+    _Subscription subscription;
+    // ignore: close_sinks
+    StreamController<R> streamController = StreamController(
+      onListen: () => _subscriptions.add(subscription),
+      onCancel: () => _subscriptions.remove(subscription),
```
Comment (arv) [inline lib/replicache.dart:370]: This seems wrong related to close. Will fix.

Comment (arv) [reply lib/replicache.dart]: I'm really not sure how the stream api is supposed to work. I think this was right after all.

Comment (arv) [reply lib/replicache.dart:373]: I'm happy enough to move on...

File: lib/replicache.dart [hunk 2]
```diff
@@ -340,10 +343,17 @@ class Replicache implements ReadTransaction {
...
-  void _fireOnChange() {
-    // TODO(arv): We should pass the ref of the current commit to ensure we get
-    // consistent results in all the subscribe callbacks.
-    _changeStreamController.add(null);
+  void _fireOnChange() async {
+    final List<_Subscription> subscriptions = _subscriptions
+        .where((s) => !s.streamController.isPaused)
+        .toList(growable: false);
+    final results = await query((tx) async {
+      final futures = subscriptions.map((s) => s.callback(tx));
```
Comment (aboodman) [inline lib/replicache.dart:351]: What happens if the callback throws?

Comment (arv) [reply lib/replicache.dart]: Look at 0bf4248ebb2910bf19b928cbf6c0328abf7ae6e0. Isn't that beautiful =P

**Pull request merged**